pkg/lockfile/helpers_test.go

```diff
@@ -31,10 +32,32 @@ func packageToString(pkg lockfile.PackageDetails) string {
 	return fmt.Sprintf("%s@%s (%s, %s)", pkg.Name, pkg.Version, pkg.Ecosystem, commit)
 }
 
-func hasPackage(packages []lockfile.PackageDetails, pkg lockfile.PackageDetails) bool {
+// checks if two strings are equal, treating any occurrences of `%%` in the
+// expected string to mean "any text"
+func areEqual(t *testing.T, actual, expect string) bool {
+	t.Helper()
+
+	expect = regexp.QuoteMeta(expect)
+	expect = strings.ReplaceAll(expect, "%%", ".+")
+
+	re := regexp.MustCompile(`^` + expect + `$`)
+
+	return re.MatchString(actual)
+}
+
+func hasPackage(t *testing.T, packages []lockfile.PackageDetails, pkg lockfile.PackageDetails) bool {
+	t.Helper()
+	// Store source here since original source is set to empty string to do equal comparison
+	pkgSource := pkg.Source
 	for _, details := range packages {
-		if details == pkg {
-			return true
+		// Custom source equality check to not be too path specific
+		// areEqual is not symmetrical, so compare both ways
+		if areEqual(t, pkgSource, details.Source) || areEqual(t, details.Source, pkgSource) {
```

Comment:
I'll dig into this more later and maybe it'll then become clear, but I would expect you shouldn't need to do any `%%`ing for `Source` (aka the path should be determinable and easy to generate for each test) so I'd say do that rather than use `%%` because it should be a more robust test.

Reply:
Hmm I'm not sure if I understand what you mean here. This is for requirements.txt because the result is an absolute path which will change depending on where the test is being ran. 

I guess we can expand `%%` to the CWD instead for more accuracy.

Reply:
You should be able to determine what that absolute path is via `filepath.Abs` - frankly we could do that in the CLI tests too, I just didn't because `%%` was available and I was being a bit lazy (and it came from the detector which needed it because it printed the db count and update time which couldn't be easily determined).

Reply:
Ah I see what you mean, I'll try that out.

Reply:
Done in a5fc1b5